src/sidebar/service-context.tsx

```diff
@@ -58,7 +58,7 @@ export const ServiceContext = createContext(fallbackInjector);
  */
 export function withServices<
   Props extends Record<string, unknown>,
-  ServiceName extends string,
+  ServiceName extends string & keyof Props,
```

Comment:
My plan was to define the type as just `keyof Props`, expecting `ServiceName` to be inferred as string considering Props has type `Record<string, unknown>`, but it ends up inferred as `string | number | Symbol` 🤷🏼‍♂️ 

Adding both `string & keyof Props` solves this problem.

Reply:
I just realized why `ServiceName` was not inferred as `string`. It's because `Props` does not have a fixed `Record<string, unknown>` type, but **extends** from it.